spring-data-mongodb/src/test/java/org/springframework/data/mongodb/core/aggregation/AccumulatorOperatorsUnitTests.java

```diff
@@ -132,6 +132,26 @@ void rendersPercentileWithExpression() {
 			.isEqualTo(Document.parse("{ $percentile: { input: [\"$scoreOne\", {\"$sum\": \"$scoreTwo\"}], method: \"approximate\", p: [0.1, 0.2] } }"));
 	}
 
+	@Test // GH-4472
+	void rendersMedianWithFieldReference() {
+
+		assertThat(valueOf("score").median().toDocument(Aggregation.DEFAULT_CONTEXT))
+				.isEqualTo(Document.parse("{ $median: { input: \"$score\", method: \"approximate\" } }"));
+
+		assertThat(valueOf("score").median().and("scoreTwo").toDocument(Aggregation.DEFAULT_CONTEXT))
+				.isEqualTo(Document.parse("{ $median: { input: [\"$score\", \"$scoreTwo\"], method: \"approximate\" } }"));
+	}
+
+	@Test // GH-4472
+	void rendersMedianWithExpression() {
+
+		assertThat(valueOf(Sum.sumOf("score")).median().toDocument(Aggregation.DEFAULT_CONTEXT))
+				.isEqualTo(Document.parse("{ $median: { input: {\"$sum\": \"$score\"}, method: \"approximate\" } }"));
+
+		assertThat(valueOf("scoreOne").median().and(Sum.sumOf("scoreTwo")).toDocument(Aggregation.DEFAULT_CONTEXT))
+				.isEqualTo(Document.parse("{ $median: { input: [\"$scoreOne\", {\"$sum\": \"$scoreTwo\"}], method: \"approximate\" } }"));
+	}
+
```

Comment:
can we add one more test using a `TypeBasedAggregationOperationContext`?
something like: 
```java
assertThat(valueOf("midichlorianCount").median().toDocument(contextFor(Jedi.class)))
    .isEqualTo("{ $median: { input: '$force', ...
```